crates/core/src/db/queries/metadatas.rs

```diff
@@ -503,79 +509,124 @@ pub fn collection_nfts(conn: &Connection, options: CollectionNftOptions) -> Resu
         )
         .take();
 
-    let mut query = Query::select()
-        .columns(vec![
-            (Metadatas::Table, Metadatas::Address),
-            (Metadatas::Table, Metadatas::Name),
-            (Metadatas::Table, Metadatas::SellerFeeBasisPoints),
-            (Metadatas::Table, Metadatas::UpdateAuthorityAddress),
-            (Metadatas::Table, Metadatas::MintAddress),
-            (Metadatas::Table, Metadatas::PrimarySaleHappened),
-            (Metadatas::Table, Metadatas::Uri),
-            (Metadatas::Table, Metadatas::Slot),
-        ])
-        .columns(vec![
-            (MetadataJsons::Table, MetadataJsons::Description),
-            (MetadataJsons::Table, MetadataJsons::Image),
-            (MetadataJsons::Table, MetadataJsons::AnimationUrl),
-            (MetadataJsons::Table, MetadataJsons::ExternalUrl),
-            (MetadataJsons::Table, MetadataJsons::Category),
-            (MetadataJsons::Table, MetadataJsons::Model),
-        ])
-        .columns(vec![(
-            CurrentMetadataOwners::Table,
-            CurrentMetadataOwners::TokenAccountAddress,
-        )])
-        .from(MetadataJsons::Table)
-        .inner_join(
-            Metadatas::Table,
-            Expr::tbl(MetadataJsons::Table, MetadataJsons::MetadataAddress)
-                .equals(Metadatas::Table, Metadatas::Address),
-        )
-        .inner_join(
-            CurrentMetadataOwners::Table,
-            Expr::tbl(Metadatas::Table, Metadatas::MintAddress).equals(
+    let uuid = Uuid::parse_str(&collection);
+    let mut query = match uuid {
+        Err(_error) => Query::select()
+            .columns(vec![
+                (Metadatas::Table, Metadatas::Address),
+                (Metadatas::Table, Metadatas::Name),
+                (Metadatas::Table, Metadatas::SellerFeeBasisPoints),
+                (Metadatas::Table, Metadatas::UpdateAuthorityAddress),
+                (Metadatas::Table, Metadatas::MintAddress),
+                (Metadatas::Table, Metadatas::PrimarySaleHappened),
+                (Metadatas::Table, Metadatas::Uri),
+                (Metadatas::Table, Metadatas::Slot),
+            ])
+            .columns(vec![
+                (MetadataJsons::Table, MetadataJsons::Description),
+                (MetadataJsons::Table, MetadataJsons::Image),
+                (MetadataJsons::Table, MetadataJsons::AnimationUrl),
+                (MetadataJsons::Table, MetadataJsons::ExternalUrl),
+                (MetadataJsons::Table, MetadataJsons::Category),
+                (MetadataJsons::Table, MetadataJsons::Model),
+            ])
+            .columns(vec![(
```

Comment:
The columns selected in the me nfts should be the same. They will have metadata, metadata json, and current owners. Its not the collection we are fetching here but the nfts.